### PR TITLE
Introduces setLayersForMap in MapService

### PR DESCRIPTION
**Pull request by KaiVolland**
This PR contains the needed code to update the list of layers contained by a mapmodule.

Posting to the MapController with a `mapModuleId` and a list of `abstractLayerIds` updates the contained layers. If the list is empty all layers will be removed.

### CODE REVIEW

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/MapService.java [hunk 1]
```diff
@@ -42,4 +50,36 @@ public void setDao(D dao) {
...
+
+	/**
+	 *
+	 * @param MapModuleId
+	 * @param abstractLayerIds
+	 * @return
+	 * @throws Exception
+	 */
+	@SuppressWarnings("unchecked")
+	public List<AbstractLayer> setLayersForMap (Integer mapModuleId, List<String> abstractLayerIds) throws Exception{
```
Comment (buehner) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/MapService.java]: The list of IDs should be of a generic type, but not String!

File: src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+	/**
+	 * 
+	 * @param moduleId
+	 * @param toolIds
+	 * @return
+	 */
+	@RequestMapping(value = "/setLayersForMap.action", method = RequestMethod.POST)
+	public @ResponseBody java.util.Map<String, Object> setLayersForMap(
+			Integer mapModuleId, @RequestParam("abstractLayerIds") List<String> abstractLayerIds) {
```
Comment (buehner) [inline src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java]: The list of IDs should be of a generic type, but not String!

**Comment by buehner** [comment]
Could you provide a short description for this PR?

**Comment by KaiVolland** [comment]
I added an description at the top now.
Didn't know why i choosed String for the IDs. I'll have a look on this before merging.

**Comment by buehner** [comment]
Is the order of the IDs of importance? If not, a `Set` would also do the job :wink:

**Comment by KaiVolland** [comment]
It is important as it currently determines the order of the layers on the map.

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/MapService.java [hunk 3]
```diff
@@ -58,4 +66,36 @@ public void setDao(D dao) {
...
+					if(layer != null){
+						layers.add(layer);
+					}
+				}
+			}
+			mapModule.setMapLayers(layers);
+			this.saveOrUpdate((E) mapModule);
+			return layers;
+		} else {
+			throw new Exception("Layer of with given Id is not a LayerGroup.");
```
Comment (dnlkoch) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/MapService.java]: Please adapt error message to map.

File: src/shogun2-web/src/main/java/de/terrestris/shogun2/web/AbstractLayerController.java [hunk 4]
```diff
@@ -83,7 +83,7 @@ protected AbstractLayerController(Class<E> type) {
 	 */
 	@RequestMapping(value = "/setLayersForLayerGroup.action", method = RequestMethod.POST)
 	public @ResponseBody Map<String, Object> setLayersForLayerGroup(
-			Integer layerGroupId, @RequestParam("abstractLayerIds") List<String> abstractLayerIds) {
+			Integer layerGroupId, @RequestParam("abstractLayerIds") List<Integer> abstractLayerIds) {
```
Comment (dnlkoch) [inline src/shogun2-web/src/main/java/de/terrestris/shogun2/web/AbstractLayerController.java]: Inconsistent use of the `@RequestParam` annotation.

File: src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+	/**
+	 * 
+	 * @param moduleId
+	 * @param toolIds
+	 * @return
+	 */
+	@RequestMapping(value = "/setLayersForMap.action", method = RequestMethod.POST)
+	public @ResponseBody java.util.Map<String, Object> setLayersForMap(
+			Integer mapModuleId, @RequestParam("abstractLayerIds") List<Integer> abstractLayerIds) {
```
Comment (dnlkoch) [inline src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java]: Inconsistent use of the `@RequestParam` annotation.

File: src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+		try {
+			List<AbstractLayer> layers = this.service.setLayersForMap(mapModuleId, abstractLayerIds);
+			return ResultSet.success(layers);
+		} catch (Exception e) {
+			return ResultSet.error("Could not set Layers for LayerGroup");
```
Comment (dnlkoch) [inline src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java]: Please adapt error message to map.

**Comment by KaiVolland** [comment]
@dnlkoch Addressed all your notes but couldn't test it on my machine.

File: src/shogun2-web/src/main/java/de/terrestris/shogun2/web/AbstractLayerController.java [hunk 7]
```diff
@@ -83,7 +82,7 @@ protected AbstractLayerController(Class<E> type) {
 	 */
 	@RequestMapping(value = "/setLayersForLayerGroup.action", method = RequestMethod.POST)
 	public @ResponseBody Map<String, Object> setLayersForLayerGroup(
-			Integer layerGroupId, @RequestParam("abstractLayerIds") List<String> abstractLayerIds) {
+			Integer layerGroupId, List<Integer> abstractLayerIds) {
```
Comment (dnlkoch) [inline src/shogun2-web/src/main/java/de/terrestris/shogun2/web/AbstractLayerController.java]: By inconsistent use I meant adding `@RequestParam` to all parameters :stuck_out_tongue_closed_eyes:

File: src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+	/**
+	 * 
+	 * @param moduleId
+	 * @param toolIds
+	 * @return
+	 */
+	@RequestMapping(value = "/setLayersForMap.action", method = RequestMethod.POST)
+	public @ResponseBody java.util.Map<String, Object> setLayersForMap(
+			Integer mapModuleId, List<Integer> abstractLayerIds) {
```
Comment (dnlkoch) [inline src/shogun2-web/src/main/java/de/terrestris/shogun2/web/MapController.java]: By inconsistent use I meant adding `@RequestParam` to all parameters :stuck_out_tongue_closed_eyes:

**Comment by dnlkoch** [comment]
Thanks, I'll merge now!

**Pull request merged**